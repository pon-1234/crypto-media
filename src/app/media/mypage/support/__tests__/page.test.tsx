/**
 * @vitest-environment jsdom
 */
import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'
import SupportPage from '../page'

// lucide-reactのモック
vi.mock('lucide-react', async () => {
  const actual = await vi.importActual('lucide-react')
  return {
    ...actual,
    ChevronLeft: () => <div data-testid="chevron-left" />,
    ChevronRight: () => <div data-testid="chevron-right" />,
    Mail: () => <div data-testid="mail-icon" />,
    MessageCircle: () => <div data-testid="message-circle-icon" />,
    Book: () => <div data-testid="book-icon" />,
  }
})

describe('SupportPage', () => {
  it('ページが正しくレンダリングされる', () => {
    const { container } = render(<SupportPage />)
    expect(container).toMatchSnapshot()
  })
}) 