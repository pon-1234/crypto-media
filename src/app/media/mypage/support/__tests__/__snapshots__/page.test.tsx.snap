// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SupportPage > ページが正しくレンダリングされる 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-50 py-8"
  >
    <div
      class="mx-auto max-w-4xl px-4 sm:px-6 lg:px-8"
    >
      <a
        class="mb-6 inline-flex items-center text-sm text-gray-600 hover:text-gray-900"
        href="/media/mypage"
      >
        <div
          data-testid="chevron-left"
        />
        マイページに戻る
      </a>
      <h1
        class="mb-8 text-2xl font-bold text-gray-900"
      >
        ヘルプ・お問い合わせ
      </h1>
      <div
        class="mb-12 grid gap-4 sm:grid-cols-3"
      >
        <a
          class="group rounded-lg bg-white p-6 shadow-sm transition-all hover:shadow-md"
          href="/media/faq"
        >
          <div
            class="mb-4 inline-flex rounded-lg bg-blue-50 p-3 text-blue-600 group-hover:bg-blue-100"
          >
            <div
              data-testid="book-icon"
            />
          </div>
          <h3
            class="mb-2 font-semibold text-gray-900"
          >
            よくある質問
          </h3>
          <p
            class="text-sm text-gray-600"
          >
            サービスに関するよくある質問をまとめています
          </p>
        </a>
        <a
          class="group rounded-lg bg-white p-6 shadow-sm transition-all hover:shadow-md"
          href="/media/glossary"
        >
          <div
            class="mb-4 inline-flex rounded-lg bg-blue-50 p-3 text-blue-600 group-hover:bg-blue-100"
          >
            <div
              data-testid="message-circle-icon"
            />
          </div>
          <h3
            class="mb-2 font-semibold text-gray-900"
          >
            用語集
          </h3>
          <p
            class="text-sm text-gray-600"
          >
            暗号資産・ブロックチェーンの専門用語を解説
          </p>
        </a>
        <a
          class="group rounded-lg bg-white p-6 shadow-sm transition-all hover:shadow-md"
          href="/media/contact"
        >
          <div
            class="mb-4 inline-flex rounded-lg bg-blue-50 p-3 text-blue-600 group-hover:bg-blue-100"
          >
            <div
              data-testid="mail-icon"
            />
          </div>
          <h3
            class="mb-2 font-semibold text-gray-900"
          >
            お問い合わせ
          </h3>
          <p
            class="text-sm text-gray-600"
          >
            その他のご質問やご要望はこちらから
          </p>
        </a>
      </div>
      <div
        class="rounded-lg bg-white p-6 shadow-sm"
      >
        <h2
          class="mb-6 text-lg font-semibold text-gray-900"
        >
          よくあるトピック
        </h2>
        <div
          class="space-y-8"
        >
          <div>
            <h3
              class="mb-3 font-medium text-gray-900"
            >
              会員登録・ログインについて
            </h3>
            <ul
              class="space-y-2"
            >
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    パスワードを忘れた場合
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    メールアドレスを変更したい
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    ログインできない
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    二段階認証の設定方法
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
            </ul>
          </div>
          <div>
            <h3
              class="mb-3 font-medium text-gray-900"
            >
              有料会員について
            </h3>
            <ul
              class="space-y-2"
            >
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    有料会員の特典
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    料金プラン
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    支払い方法の変更
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    解約・退会について
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
            </ul>
          </div>
          <div>
            <h3
              class="mb-3 font-medium text-gray-900"
            >
              記事・コンテンツについて
            </h3>
            <ul
              class="space-y-2"
            >
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    記事が読めない
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    記事の誤りを報告したい
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    記事のリクエスト
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
              <li>
                <a
                  class="flex items-center justify-between rounded-md px-3 py-2 text-sm text-gray-600 hover:bg-gray-50 hover:text-gray-900"
                  href="/media/faq"
                >
                  <span>
                    お気に入り機能の使い方
                  </span>
                  <div
                    data-testid="chevron-right"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="mt-8 rounded-lg bg-blue-50 p-6 text-center"
      >
        <p
          class="mb-4 text-gray-700"
        >
          お探しの情報が見つからない場合は、お気軽にお問い合わせください
        </p>
        <a
          class="inline-flex items-center rounded-md bg-blue-600 px-6 py-3 text-white hover:bg-blue-700"
          href="/media/contact"
        >
          <div
            data-testid="mail-icon"
          />
          お問い合わせフォームへ
        </a>
      </div>
    </div>
  </div>
</div>
`;
